### PR TITLE
syslog & ramlog update

### PR DESCRIPTION
**Pull request by GUIDINGLI**
## Summary

syslog & ramlog: add BOARDIOC_RESET_CAUSE for syslog & ramlog

## Impact

syslog & ramlog

## Testing

VELA

### CODE REVIEW

**Comment by raiden00pl** [comment]
commit message for `8eb385f` seems unrelated to its content. @GUIDINGLI could you take a look at this?

File: drivers/syslog/ramlog.c [hunk 1]
```diff
@@ -235,6 +240,17 @@ static void ramlog_initbuf(void)
       return;
     }
 
+#ifdef CONFIG_BOARDCTL_RESET_CAUSE
+  memset(&cause, 0, sizeof(cause));
+  ret = boardctl(BOARDIOC_RESET_CAUSE, (uintptr_t)&cause);
+  if (ret >= 0 && !cause.cause && !cause.flag)
```
Comment (xiaoxiang781216) [inline drivers/syslog/ramlog.c:246]:
```suggestion
  if (ret >= 0 && cause.cause  != BOARDIOC_RESETCAUSE_SYS_CHIPPOR)
```

File: drivers/syslog/syslog_rpmsg.c [hunk 2]
```diff
@@ -395,6 +400,16 @@ void syslog_rpmsg_init_early(FAR void *buffer, size_t size)
   priv->buffer = buffer;
   priv->size   = size;
 
+#ifdef CONFIG_BOARDCTL_RESET_CAUSE
+  memset(&cause, 0, sizeof(cause));
+  ret = boardctl(BOARDIOC_RESET_CAUSE, (uintptr_t)&cause);
+  if (ret >= 0 && !cause.cause && !cause.flag)
```
Comment (xiaoxiang781216) [inline drivers/syslog/syslog_rpmsg.c:407]:
```suggestion
  if (ret >= 0 && cause.cause  != BOARDIOC_RESETCAUSE_SYS_CHIPPOR)
```

File: drivers/syslog/ramlog.c [hunk 3]
```diff
@@ -235,6 +240,17 @@ static void ramlog_initbuf(void)
       return;
     }
 
+#ifdef CONFIG_BOARDCTL_RESET_CAUSE
+  memset(&cause, 0, sizeof(cause));
+  ret = boardctl(BOARDIOC_RESET_CAUSE, (uintptr_t)&cause);
+  if (ret >= 0 && cause.cause != BOARDIOC_RESETCAUSE_SYS_CHIPPOR)
```
Comment (xiaoxiang781216) [inline drivers/syslog/ramlog.c:246]: should we check the equal here?

**Pull request merged**